### PR TITLE
Move CI from Travis to Github Actions

### PR DESCRIPTION
**Pull request by proux01**
see https://coq.zulipchat.com/#narrow/stream/237663-coq-community-devs.20.26.20users/topic/New.20Travis.20limits.2E for context

@erikmd @Zimmi48 you are probably the most qualified here

### CODE REVIEW

File: .github/workflows/build-coq-bignums.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    strategy:
+      matrix:
+        image:
+          - 'coqorg/coq:dev'
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - uses: coq-community/docker-coq-action@v1
+        with:
+          opam_file: 'coq-bignums.opam'
```
Comment (erikmd) [inline .github/workflows/build-coq-bignums.yml:24]: @proux01 thanks for opening this PR.
The workflow needs to be extended to run the same tests as the Travis CI config:
```suggestion
      - uses: coq-community/docker-coq-action@v1
        with:
          opam_file: 'coq-bignums.opam'
          custom_image: ${{ matrix.image }}
          custom_script: |
            startGroup Print opam config
              opam config list; opam repo list; opam list
            endGroup
            startGroup Build dependencies
              opam pin add -n -y -k path $PACKAGE $WORKDIR
              opam update -y
              opam install -y -j 2 $PACKAGE --deps-only
            endGroup
            startGroup Build
              opam install -y -v -j 2 $PACKAGE
              opam list
            endGroup
            startGroup Workaround permission issue
              sudo chown coq:coq .
            endGroup
            startGroup Run tests
              make -C tests
            endGroup
            startGroup Uninstallation test
              opam remove $PACKAGE
            endGroup
      - name: Revert permissions
        if: ${{ always() }}
        run: sudo chown 1001:116 .
```
Admittedly, this is a bit verbose (albeit less than with Travis :), but we'll be able to refactor/simplify this as soon as https://github.com/coq-community/docker-coq-action/issues/22#issuecomment-695803544 is implemented!

File: .github/workflows/build-coq-bignums.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '**'
+
```
Comment (erikmd) [inline .github/workflows/build-coq-bignums.yml:10]: BTW @Zimmi48 I've just [disabled the Travis CI cron build](https://travis-ci.com/github/coq/bignums/settings).
So @proux01 you might want to re-add some daily build in the workflow:

```suggestion
  schedule:
  # test master every day at 05:00 UTC
    - cron: '0 5 * * *'

```

File: .github/workflows/build-coq-bignums.yml (hunk 1 above)
Comment (proux01) [reply .github/workflows/build-coq-bignums.yml:24]: Thanks, I wouldn't have found this myself.

**Comment by erikmd** [comment]
@proux01 the PR seems very OK but as it's opened from a fork we won't be able to test it before merging :)
@Zimmi48 I let you take another look and merge it?

**Review by Zimmi48 — APPROVED** [review]
LGTM!
What is special in the custom script? The tests step and the remove step? Should we extend the template to support this?

**Comment by Zimmi48** [comment]
The CI run after merging this PR failed during the test-suite run:

    Error: System error: "./success/.NumberScopes.aux: Permission denied"

See https://github.com/coq/bignums/runs/1438451310?check_suite_focus=true

**Comment by erikmd** [comment]
> What is special in the custom script? The tests step and the remove step?

Only the tests step and (because of the tests step) the fix-permissions step

> Should we extend the template to support this?

maybe yes, but before that we should just implement https://github.com/coq-community/docker-coq-action/issues/22#issuecomment-695803544

**Comment by erikmd** [comment]
> The CI run after merging this PR failed during the test-suite run:
> 
> ```
> Error: System error: "./success/.NumberScopes.aux: Permission denied"
> ```
> 
> See https://github.com/coq/bignums/runs/1438451310?check_suite_focus=true

Indeed, I guess this comes from a missing `-R` flag to `chown`. I will push a patch, stay tuned…

**Comment by erikmd** [comment]
Ci is green :heavy_check_mark: https://github.com/coq/bignums/actions/runs/377413533

**Comment by Zimmi48** [comment]
Great!

**Comment by erikmd** [comment]
Regarding https://github.com/coq-community/docker-coq-action/issues/22#issuecomment-695803544 I can have a look, but not immediately… Which ETA would you like for it?

**Comment by Zimmi48** [comment]
No strong urgency. I guess we'll just patch the config for the projects that need it.

**Comment by erikmd** [comment]
OK @Zimmi48 ! so in the meantime, bignums' [.yml](https://github.com/coq/bignums/blob/master/.github/workflows/build-coq-bignums.yml) file could serve as a running example…

**Pull request merged**